pkg/scheduler/predicates/ha.go

```diff
@@ -469,3 +474,22 @@ func isPodDesired(tc *v1alpha1.TidbCluster, component, podName string) bool {
 	}
 	return ordinals.Has(ordinal)
 }
+
+func isFailureMember(tc *v1alpha1.TidbCluster, component, podName string) bool {
+	if component == v1alpha1.PDMemberType.String() {
+		for _, fm := range tc.Status.PD.FailureMembers {
+			if fm.PodName == podName {
+				return true
+			}
+		}
+		return false
+	}
+
+	for _, fs := range tc.Status.TiKV.FailureStores {
+		if fs.PodName == podName {
+			return true
```

Comment:
Also need to check the current store status, because the failurerStores will not be removed automatically.